packages/atomic-react/src/components/SearchInterfaceWrapper.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useRef} from 'react';
+import type {JSX} from '@coveo/atomic';
+import {
+  buildSearchEngine,
+  getSampleSearchEngineConfiguration,
+} from '@coveo/atomic/headless';
+import {AtomicSearchInterface} from './stencil-generated/index';
+
+type ExecuteSearch = HTMLAtomicSearchInterfaceElement['executeFirstSearch'];
+interface WrapperProps extends JSX.AtomicSearchInterface {
+  onReady?: (executeFirstSearch: ExecuteSearch) => Promise<void>;
+  theme?: string | 'none';
+}
+
+const DefaultProps: Required<
+  Pick<WrapperProps, 'onReady' | 'theme' | 'engine'>
+> = {
+  onReady: (executeFirstSearch) => {
+    return executeFirstSearch();
+  },
+  theme: 'coveo',
+  engine: buildSearchEngine({
+    configuration: getSampleSearchEngineConfiguration(),
+  }),
+};
+
+export const AtomicSearchInterfaceWrapper = (
+  props: React.PropsWithChildren<WrapperProps>
+) => {
+  const mergedProps = {...DefaultProps, ...props};
+  const {engine, theme, onReady, ...allOtherProps} = mergedProps;
+  const searchInterfaceRef = useRef<HTMLAtomicSearchInterfaceElement>(null);
+  if (theme !== 'none') {
+    import(`@coveo/atomic/dist/atomic/themes/${theme}.css`);
+  }
+
+  useEffect(() => {
```

Comment:
Looking at this reminds me: do we need / should we add a cleanup/unsubscribe function to `atomic-search-interface`? I wonder if that would be needed for single page applications.